File: app/src/main/java/vn/ditagis/com/tanhoa/qlsc/ShowCaptureActivity.java
package vn.ditagis.com.tanhoa.qlsc;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.RequiresApi;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.ImageView;

import vn.ditagis.com.tanhoa.qlsc.entities.DApplication;

public class ShowCaptureActivity extends AppCompatActivity {
    private DApplication mApplication;

    @RequiresApi(api = Build.VERSION_CODES.O)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_show_capture);
        mApplication = (DApplication) getApplication();


        byte[] capture = mApplication.capture;
        if (capture != null) {
            ImageView imageView = findViewById(R.id.show_capture_imageView);
            Bitmap decodeBitmap = BitmapFactory.decodeByteArray(capture, 0, capture.length);
            imageView.setImageBitmap(decodeBitmap);
        }
    }

    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.show_capture_cancel:
                goHomeCancel();
                break;
            case R.id.show_capture_ok:
                goHome();
                break;
        }
    }

    @Override
    public void onBackPressed() {
        goHomeCancel();
    }


    public void goHome() {
        Intent intent = new Intent();
        setResult(RESULT_OK, intent);
        finish();
    }

    public void goHomeCancel() {
        Intent intent = new Intent();
        setResult(RESULT_CANCELED, intent);
        finish();
    }
}

File: app/src/main/java/vn/ditagis/com/tanhoa/qlsc/ShowCaptureActivity.kt
package vn.ditagis.com.tanhoa.qlsc

import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.Window
import android.view.WindowManager
import kotlinx.android.synthetic.main.activity_show_capture.*

import vn.ditagis.com.tanhoa.qlsc.entities.DApplication

class ShowCaptureActivity : AppCompatActivity() {
    private var mApplication: DApplication? = null

    @RequiresApi(api = Build.VERSION_CODES.O)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_show_capture)
        mApplication = application as DApplication


        val capture = mApplication!!.capture
        if (capture != null) {
            val decodeBitmap = BitmapFactory.decodeByteArray(capture, 0, capture.size)
            show_capture_imageView.setImageBitmap(decodeBitmap)
        }
    }

    fun onClick(view: View) {
        when (view.id) {
            R.id.show_capture_cancel -> goHomeCancel()
            R.id.show_capture_ok -> goHome()
        }
    }

    override fun onBackPressed() {
        goHomeCancel()
    }


    private fun goHome() {
        val intent = Intent()
        setResult(RESULT_OK, intent)
        finish()
    }

    private fun goHomeCancel() {
        val intent = Intent()
        setResult(RESULT_CANCELED, intent)
        finish()
    }
}
